api/include/opentelemetry/plugin/factory.h

```diff
@@ -36,8 +36,9 @@ class Factory final
    * @param error_message on failure this will contain an error message.
    * @return a Tracer on success or nullptr on failure.
    */
-  std::shared_ptr<Tracer> MakeTracer(nostd::string_view tracer_config,
-                                     std::string &error_message) const noexcept
+  std::shared_ptr<opentelemetry::trace::Tracer> MakeTracer(nostd::string_view tracer_config,
```

Comment:
This method is mixing **std** and **nostd** classes. Assuming that this is implementation is NOT ABI compatible across different compilers having **std::shared_ptr** for return value and **std::string** for error message on signature, is there a practical reason why we require the tracer_config be of **nostd** type?

Reply:
Our [ABI policy](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/docs/abi-policy.md) says this:
> Abstract classes in the OpenTelemetry API must use ABI stable types in their virtual function signatures.

`Factory` is not an abstract class and `MakeTracer` is not a virtual method, so there's no need for `nostd` types.

I think the reason for `tracer_config` being a `string_view` is that it makes it more efficient to pass in `std::string` (without need to copy) and `const char*` (without need to create a `std::string`). But it's not an ABI requirement in this case.